app/controllers/facility_accounts_controller.rb

```diff
@@ -70,7 +47,32 @@ def show
   # GET /facilities/:facility_id/accounts/new
   def new
     @owner_user = User.find(params[:owner_user_id])
-    @account    = @owner_user.accounts.new(:expires_at => Time.zone.now + 1.year)
+    @available_account_types = available_account_types
+    @current_account_type = current_account_type
+    @account = current_account_type.constantize.new
+  end
+
+  # POST /facilities/:facility_id/accounts
+  def create
+    @owner_user = User.find(params[:owner_user_id])
+    @available_account_types = available_account_types
+    @current_account_type = current_account_type
+
+    @account = AccountBuilder.for(params[:account_type]).new({
+      account_type: params[:account_type],
```

Comment:
Passing account type to both seemed redundant to me as well.  But multiple account types may use the same builder, so given an account builder class object we shouldn't try to assume an account type.  

We could have the factory return an instance of the account builder with the account type already stored.  And either pass the initialization args to the factory method as well or support setting the additional args with another `new`-like method.  But I felt that the instance factory approach makes it less clear what's going on.  What do you think?